components/infra-proxy-service/storage/postgres/postgres.go

```diff
@@ -59,10 +60,16 @@ func parsePQError(e *pq.Error) error {
 	case "unique_violation":
 		return storage.ErrConflict
 	case "foreign_key_violation":
-		// in this piece of code, a foreign key violation means the server the org
-		// should be added to doesn't exist; a "not found" seems like an OK
-		// approximation for now
-		return storage.ErrNotFound
+		// In this piece of code, a foreign key violation means the server the org
+		// should be added to doesn't exist.
+		// The only way to distinguish this violation by comparing the server_id key in Error detail.
+		// DETAIL:  Key (server_id)=(2e831dec-f25c-4fde-aef6-8f68ea366251) is not present in table "servers".
+		if strings.Contains(e.Detail, "(server_id)") {
```

Comment:
I am fully sold on this comparison but I don't have a better solution as of now.

Reply:
That sounds reasonable, but let us tighten it up a bit, also confirming the message ("not present") and the table ("servers"), so I think a regex check like this would be preferable. Note that we want to match word boundaries (`\<` and `\>`) to make sure those are not just substrings.
```
regexp.MustCompile('\<server_id\>.*is not present.*table.*\<servers\>').MatchString(e.Detail)
```